src/src/Helpers/Networking.cpp

```diff
@@ -1682,6 +1682,115 @@ int http_authenticate(const String& logIdentifier,
+            // Extract the parameters from the URL
+            int start = url.indexOf(eventName + '=');
+            if (start == -1) {
+                return; // No parameters found for the given eventName
+            }
+            start += eventName.length() + 1;
+            const int end = url.indexOf('&', start);
+            const String params = (end == -1) ? url.substring(start) : url.substring(start, end);
+
+            if (!params.isEmpty()) {
+                String keys[20];
+                int keyCount = 0;
+                int startIndex = 0;
+                int commaIndex = params.indexOf(',');
+
+                // Split and add keys to the array
+                while (commaIndex != -1) {
+                    if (keyCount >= 20) {
+                        // Stop adding keys if array is full
+                        addLog(LOG_LEVEL_ERROR, F("Too many keys in the URL"));
+                        break;
+                    }
+                    String key = params.substring(startIndex, commaIndex);
+                    keys[keyCount++] = key;
+                    startIndex = commaIndex + 1;
+                    commaIndex = params.indexOf(',', startIndex);
+                }
+
+                // Add the last key
+                if (keyCount < 20) {
+                    const String lastKey = params.substring(startIndex);
+                    keys[keyCount++] = lastKey;
+                }
+
+                String csv;
+                const int startStringIndex = str.indexOf("\"" + eventName + "\":") + eventName.length() + 4;
+                const int endStringIndex = str.indexOf("}", startStringIndex);
+
+                for (int i = 0; i < keyCount; i++) // Use keyCount to limit the iteration
+                {
+                    String key = keys[i];
+                    String value;
+                    int startIndex = str.indexOf(key + "\":", startStringIndex);
+
+                    if (startIndex == -1) {
+                        // Handle case where key is not found
+                        value = F("-256"); // Placeholder value
+                    } else {
+                        int endIndex = 0;
+
+                        if (!equals(eventName, F("current"))) {
+                            // In daily and hourly, the values are stored in an array
+                            startIndex += key.length() + 3; // Move index past the key
+                            endIndex = str.indexOf("]", startIndex);
+                        } else {
+                            startIndex += key.length() + 2; // Move index past the key
+                            endIndex = str.indexOf(",", startIndex);
+                        }
+
+                        // Find the index of the next comma
+                        if ((endIndex == -1) || (endIndex > str.indexOf("}", startIndex))) {
+                            endIndex = str.indexOf("}", startIndex); // If no comma is found or comma comes after },

```

Comment:
Ditto

Reply:
This too i guess right?:
`int startIndex = str.indexOf(key + "\":", startStringIndex);`

Reply:
Here you can use `...strformat(F("%s\":"), key.c_str()),,,`

Reply:
or `...concat(key, F("\":"))...`